src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -148,18 +148,10 @@ object LoadVCF {
     (TStruct(fb.result()), canonicalFlags)
   }
 
-  def parseHeader(hc: HailContext, reader: HtsjdkRecordReader, file: String): VCFHeaderInfo = {
-    val hConf = hc.hadoopConf
-
-    val headerLines = hConf.readFile(file) { s =>
-      Source.fromInputStream(s)
-        .getLines()
-        .takeWhile { line => line(0) == '#' }
-        .toArray
-    }
+  def parseHeader(reader: HtsjdkRecordReader, lines: Array[String]): VCFHeaderInfo = {
```

Comment:
should take an Iterator[String], not Array[String] - there's no reason to construct an array for anything but the first (loaded locally)